cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/linkanalysis/PageRank.scala

```diff
@@ -0,0 +1,70 @@
+package com.twitter.cassovary.algorithms.linkanalysis
+
+import com.twitter.cassovary.graph.{GraphDir, Node, DirectedGraph}
+import com.twitter.cassovary.util.Progress
+
+/**
+ * Parameters for PageRank
+ * @param dampingFactor Probability of NOT randomly jumping to another node
+ * @param maxIterations The maximum number of times that the link analysis algorithm will run before termination
+ * @param tolerance The maximum error allowed.
+ */
+case class PageRankParams(dampingFactor: Double = 0.85,
+                          override val maxIterations: Option[Int] = Some(10),
+                          override val tolerance: Double = 1e-8)
+  extends Params(maxIterations, tolerance)
+
+/**
+ * An object containing information to fully describe a page rank iteration.
+ * @param pageRank The current set of pageRank values
+ * @param error  The T1 error for the current iteration vs the previous iteration.
+ */
+case class PageRankIteration(pageRank: Array[Double],
+                             override val error: Double,
+                             override val iteration: Int)
+  extends Iteration(error, iteration)
+
+/**
+ * PageRank is a link analysis algorithm designed to measure the importance of nodes in a graph.
+ * Popularized by Google.
+ *
+ * Unoptimized for now, and runs in a single thread.
+ */
+case class PageRank(graph: DirectedGraph[Node], params: PageRankParams)
```

Comment:
case classes are great for small classes such as Iteration or Params, but I don't generally use case classes unless there is a clear reason to for classes with more code / elements.